wit/tcp.wit

```diff
@@ -43,29 +50,43 @@ default interface tcp {
 	/// - the socket is transitioned into the Connection state
 	/// - a pair of streams is returned that can be used to read & write to the connection
 	/// 
-	/// Fails when:
-	/// - the socket is already bound to a different network.
-	/// - the provided network does not allow connections to the specified endpoint.
-	/// - the socket is already in the Connection or Listener state.
-	/// - either the remote IP address or port is 0.
-	/// 
-	/// References
+	/// # Typical errors
+	/// - `timeout`:                   Connection timed out. (ETIMEDOUT)
+	/// - `connection-refused`:        The connection was forcefully rejected. (ECONNREFUSED)
+	/// - `connection-reset`:          The connection was reset. (ECONNRESET)
+	/// - `remote-unreachable`:        The remote address is not reachable. (EHOSTUNREACH, EHOSTDOWN, ENETUNREACH, ENETDOWN)
+	/// - `address-family-mismatch`:   The `remote-address` has the wrong address family. (EAFNOSUPPORT)
+	/// - `invalid-remote-address`:    The IP address in `remote-address` is set to INADDR_ANY (`0.0.0.0` / `::`). (EADDRNOTAVAIL on Windows)
+	/// - `invalid-remote-address`:    The port in `remote-address` is set to 0. (EADDRNOTAVAIL on Windows)
+	/// - `already-attached`:          The socket is already attached to a different network. The `network` passed to `connect` must be identical to the one passed to `bind`.
+	/// - `already-connected`:         The socket is already in the Connection state. (EISCONN)
+	/// - `already-listening`:         The socket is already in the Listener state. (EOPNOTSUPP, EINVAL on Windows)
+	/// - `ephemeral-ports-exhausted`: Tried to perform an implicit bind, but there were no ephemeral ports available. (EADDRINUSE, EADDRNOTAVAIL on Linux, EAGAIN on BSD)
+	/// - `concurrency-conflict`:      Another `bind`, `connect` or `listen` operation is already in progress. (EALREADY)
```

Comment:
What's the mapping for `EINPROGRESS`?

Reply:
There isn't one. :)

I'm planning a follow-up PR that fixes some current design issues surrounding non-blocking functions. Especially `connect`, which is broken in its current form. (Because you can't recover the input/output-stream handles)

As matter of fact, I intend to drop blocking mode entirely and let libc handle that.